### PR TITLE
AppBar refresh - Site Settings, Comments, People, Plugins

### PR DESCRIPTION
**Pull request by khaykov**
#12608 needs to be merged first.

This PR updates AppBar design and adds lift on scroll functionality to the following screens (and related nested screens):

- Site Settings
- Comments
- People
- Plugins

What is lift on scroll? When you scroll content below the AppBar its elevation changes, and as a result shadow appears in light mode, and in Dark Mode color of AppBar gets lighter.

@osullivanchris I would like your opinion on changes to the filter element in Comments and People screens. I updated the look of the filtered list according to your mock, but there is a couple of functional limitations that I was not able to overcome without rewriting the whole component:
- Filter is attached to the top of the list, and will not enter unless you scroll to the top of the list (otherwise it will look weird without having any elevation or background color for filter).
- Even where there is not enough content in the list to scroll, the filter will scroll under AppBar (this is preexisting behavior).
- Scrolling filter under AppBar will not trigger elevation change, as shown on this gif:
[Image from Gyazo](https://gyazo.com/89c0b52929d3ac4c8412c5c6078d8019)

[Image from Gyazo](https://gyazo.com/a4eb46a213dae4c18f19964227f2e373)

To test:

For all screens:

Make sure nothing looks out of place (including ActionMode) :)
Make sure the toolbar state is restored on configuration change.

Test 1:

- Navigate to the Site Settings screen
- Scroll the preference list and make sure the lift on scroll works (add a couple of entries and rotate the screen into landscape mode to make the list scrollable).
- Check Tags screen and make sure that lift on scroll works
- Navigate to nested preference screen by clicking on `More` under `Discussion` category.
- Make sure that the lift on scroll in nested preference screen works as expected.
- Check Blacklist and Hold for moderation screen under `Discussion` category and make sure lift on scroll there works as expected (add a couple of entries and rotate the screen into landscape mode to make the list scrollable).

Test 2:

- Navigate to Comments screen.
- Navigate to comment details (make sure you have at least one long comment).
- Make sure the lift on scroll works when you scroll the comment.
- Swipe comments sideways (they are in ViewPager) and make sure the toolbar elevation changes based on the scrolled state of each comment.

Test 3:

- Navigate to People screen.
- Scroll the content and make sure the lift on scroll works.
- Navigate to Add Person screen, rotate your device into horizontal orientation, and make sure the lift on scroll works when you scroll the form.

Test 4:

- Navigate to Plugins screen.
- Scroll the content and make sure the lift on scroll works (you might need to rotate your device).
- Open plugin list by tapping Manage or See All buttons above plugin rows.
- Scroll the content and make sure the lift on scroll works (you might need to rotate your device).

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by khaykov** [comment]
> I'm not sure I understand this comment, sorry. Was the word 'enter' correct? It works as I expect though. I don't think it would be appropriate to make it sticky/fixed. And it would create confusion with headers and elevation. Maybe that covers the question!

Yep, it does :) 

> This was something I noticed. And I was going to request a change before I saw you covered it. I would have expected the elevation to occur immediately, rather than only when the filter is scrolled out of view. Is it a limitation that we are stuck with?

That component is a bit tricky to work with, but I think I have an idea of how to make it work. I'll get to it asap.

**Comment by khaykov** [comment]
@osullivanchris So I did some more research regardin this:

> This was something I noticed. And I was going to request a change before I saw you covered it. I would have expected the elevation to occur immediately, rather than only when the filter is scrolled out of view. Is it a limitation that we are stuck with?

And in short, yes this is a limitation we are stuck with - AppBar changes elevation when the content of the list scroll under it, and since the filter bar is not a part of the list it's not triggering the elevation change. Hope this makes sense. Another way we can approach this is like this:

[Image from Gyazo](https://gyazo.com/2b34a9e0dfad2ed3994fca16e2c0a00d)

**Comment by osullivanchris** [comment]
Thanks for sharing those @khaykov! I think the original is best to stick with. Its a small detail. And it takes a sharp eye to notice it. I think the other options are more jarring, but it was good to explore alternatives. 

Let's push forward with what you originally had.

**Comment by khaykov** [comment]
@renanferrari Thank you for taking a look! I did some digging and noticed that the issue with the toolbar in Site Settings happens when there's a fling when you scroll up (if you scroll slowly the lift on scroll works as expected). I tried a couple of things but wasn't able to fix it. I'm pretty sure the lift on scroll will work as expected after we migrate to androidx version of preferences since it's using RecyclerView which works way better with AppBar layout.

I fixed the double appbar and color of action mode icons.

I also checked the action toolbar that appears above the appbar, and looks like it was intentional - we do some dark magic with appbar in preferences, and I guess at the time there was no way to show action mode in the "fake" appbar in preference dialog. Maybe this is something we can also fix after migrating to new preferences.

File: WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginBrowserActivity.java [hunk 1]
```diff
@@ -227,6 +231,17 @@ public boolean onOptionsItemSelected(final MenuItem item) {
...
     }
 
+    @Override
+    public void onBackPressed() {
+        if (getSupportFragmentManager().getBackStackEntryCount() > 0) {
+            mAppBar.post(() -> {
+                mAppBar.setLiftOnScrollTargetViewId(R.id.scroll_view);
+                mAppBar.requestLayout();
+            });
+        }
```
[the remark this answers is not in the document]
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginBrowserActivity.java]: Added comment and extracted the logic into an extension 👍

**Pull request merged**